### PR TITLE
Week3 pull1

### PR DESCRIPTION
**Pull request by runyans7**

### CODE REVIEW

File: app/controllers/queue_items_controller.rb [hunk 1]
```diff
@@ -14,11 +14,31 @@ def create
...
     queue_item = QueueItem.find(params[:id])
     queue_item.destroy if current_user.queue_items.include?(queue_item)
+    current_user.normalize_positions
+    redirect_to my_queue_path
+  end
+
+  def update_queue
+    begin
+      update_queue_items
+      current_user. normalize_positions
```
Comment (tomtomecek) [inline app/controllers/queue_items_controller.rb]: Why there is a space between `.` and `normalize_positions` ?

File: app/models/user.rb [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
   has_many :reviews
-  has_many :queue_items
+  has_many :queue_items,  -> { order(:position) }
 
   validates_presence_of :email, :password, :fullname
   validates_uniqueness_of :email
   has_secure_password validations: false
+
+  def normalize_positions
+    self.queue_items.each_with_index do |queue_item, index|
```
Comment (tomtomecek) [inline app/models/user.rb]: There is no need for `self.` :)

File: app/models/user.rb [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
+      queue_item.update_attributes(position: index + 1)
```
Comment (tomtomecek) [inline app/models/user.rb]: You can use `update` instead of `update_attributes`.

File: app/models/user.rb (hunk 2 above)
Comment (runyans7) [inline app/models/user.rb]: Thanks. I am still a little fuzzy on when self is require and I tend to over use it.

Comment (tomtomecek) [inline app/models/user.rb]: Rather overuse it than not, but eventually with more experience you will get more feeling into your hands and eyes.

**Pull request merged**